src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,108 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_ARRAY_SIZE = 10;
+    private Object[] innerArray = new Object[DEFAULT_ARRAY_SIZE];
+    private int size = 0;
+
     @Override
     public void add(T value) {
-
+        growIfArrayFull();
+        innerArray[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index != size) {
+            checkIndex(index);
+        }
+        growIfArrayFull();
+        if (size == 0) {
+            add(value);
+        } else if (size == 1) {
+            innerArray[1] = innerArray[0];
+            innerArray[0] = value;
+            size++;
+        } else {
+            System.arraycopy(innerArray, index, innerArray, index + 1, size - index);
+            innerArray[index] = value;
+            size++;
```

Comment:
you can simplify this method like this 
```suggestion
    checkIndex(index);
    growIfArrayFull();
    System.arraycopy(innerArray, index, innerArray, index + 1, size - index);
    innerArray[index] = value;
        size++;
```